src/test/java/com/epam/tc/hw5/steps/UserTableSteps.java

```diff
@@ -0,0 +1,109 @@
+                .as("Incorrect page opened")
+                .isEqualTo(userTablePage.getUserTablePageURL());
+        softly.assertAll();
+    }
+
+    @Then("{string} Number Type Dropdowns should be displayed on Users Table on User Table Page")
+    public void numberOfTypeDropdownsIsDisplayed(String number) {
+        softly.assertThat(userTablePage.getNumberTypeDropdownsSize())
+                .as("Incorrect number of Type Dropdowns")
+                .isEqualTo(Integer.parseInt(number));
+        softly.assertAll();
+    }
+
+    @Then("{string} Usernames should be displayed on Users Table on User Table Page")
+    public void numberOfUsernamesIsDisplayed(String number) {
+        softly.assertThat(userTablePage.getUserNamesNumber())
+                .as("Incorrect number of Type Dropdowns")
+                .isEqualTo(Integer.parseInt(number));
+        softly.assertAll();
+    }
+
+    @Then("{string} Description texts under images should be displayed on Users Table on User Table Page")
+    public void numberOfDescriptionTextsIsDisplayed(String number) {
+        softly.assertThat(userTablePage.getDescriptionTextsNumber())
+                .as("Incorrect number of Type Dropdowns")
+                .isEqualTo(Integer.parseInt(number));
+        softly.assertAll();
+    }
+
+    @Then("{string} checkboxes should be displayed on Users Table on User Table Page")
+    public void numberOfCheckboxesIsDisplayed(String number) {
+        softly.assertThat(userTablePage.getNumberOfCheckboxes())
+                .as("Incorrect number of Type Dropdowns")
+                .isEqualTo(Integer.parseInt(number));
+        softly.assertAll();
+    }
+
+    @Then("User table should contain following values:")
+    public void user_table_should_contain_following_values(io.cucumber.datatable.DataTable dataTable) {
+        List<List<String>> userTable = dataTable.cells().subList(1, 7);
+        List<String> number = new ArrayList<>();
+        List<String> user = new ArrayList<>();
+        List<String> description = new ArrayList<>();
+        for (List<String> element : userTable) {
+            number.add(element.get(0));
+            user.add(element.get(1));
+            description.add(element.get(2));
+        }
+        softly.assertThat(userTablePage.getNumberFromTable())
+                .as("Incorrect numbers")
+                .isEqualTo(number);
+        softly.assertThat(userTablePage.getUserFromTable())
+                .as("Incorrect users")
+                .isEqualTo(user);
+        softly.assertThat(userTablePage.getDescriptionFromTable())
+                .as("Incorrect description")
+                .isEqualTo(description);
+        softly.assertAll();
+    }
+
+    @Then("droplist should contain values in column Type for user {string}")
+    public void droplistContainValuesInColumn(String userName, io.cucumber.datatable.DataTable dataTable) {
+        List<List<String>> valuesTable = dataTable.cells().subList(1, 4);
```

Comment:
Are you sure about magic numbers 1 and 4? Why 4? If the table contains 100 values will your method work correctly?

Reply:
Fixed